### PR TITLE
Hide empty accounts from search results

### LINKED ISSUE #446: [Bug]: Don't show account search result if returned account is empty
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SUMMARY

_No response_

### STEPS TO REPRODUCE

https://explorer.dev.oasis.io/search?q=0xBA504818FdD8D3dBA2Ef8fD9B4F4D5c71aD1d1D3
image

### EXPECTED BEHAVIOR

_No response_

### PR DESCRIPTION
**Pull request by csillag**
The first commit simplifies / centralizes things a little bit.
The second commit fixes #446.

### CODE REVIEW

File: src/app/pages/SearchResultsPage/hooks.ts [hunk 1]
```diff
@@ -84,3 +94,33 @@ export function useRuntimeAccountConditionally(
     results: queries.map(query => query.data?.data).filter(isDefined),
   }
 }
+
+export const useSearch = () => {
+  const q = useParamSearch()
```
Comment (lukaw3d) [inline src/app/pages/SearchResultsPage/hooks.ts:99]: I would move this out of useSearch

File: src/app/pages/SearchResultsPage/hooks.ts [hunk 2]
```diff
@@ -84,3 +94,33 @@ export function useRuntimeAccountConditionally(
...
+  const accounts = [
+    ...(queries.oasisAccount.results || []),
+    ...(queries.evmBech32Account.results || []),
+  ].filter(isAccountNonEmpty)
+  const allResults = [...blocks, ...transactions, ...accounts]
+  const results: SearchResults = {
+    blocks,
+    transactions,
+    accounts,
+    combined: allResults,
```
Comment (lukaw3d) [inline src/app/pages/SearchResultsPage/hooks.ts:120]: i would rename to allResults

File: src/oasis-indexer/api.ts [hunk 3]
```diff
@@ -44,6 +44,16 @@ declare module './generated/api' {
   }
 }
 
+export const isAccountEmpty = (account: RuntimeAccount) => {
```
Comment (lukaw3d) [inline src/oasis-indexer/api.ts:47]: Commit name `Define empty test for accounts` => `Define empty check for accounts` or `Define isAccountEmpty` :shrug: (reserve "test" for tests)

File: src/app/pages/SearchResultsPage/hooks.ts (hunk 1 above)
Comment (csillag) [reply src/app/pages/SearchResultsPage/hooks.ts]: Done

File: src/app/pages/SearchResultsPage/hooks.ts (hunk 2 above)
Comment (csillag) [reply src/app/pages/SearchResultsPage/hooks.ts]: Done

File: src/oasis-indexer/api.ts (hunk 3 above)
Comment (csillag) [reply src/oasis-indexer/api.ts:47]: Done

File: src/app/pages/SearchResultsPage/hooks.ts [hunk 4]
```diff
@@ -7,21 +7,31 @@ import {
...
+import { SearchParams, useParamSearch } from '../../components/Search/search-utils'
 
 function isDefined<T>(item: T): item is NonNullable<T> {
   return item != null
 }
 
 export type ConditionalResults<T> = { isLoading: boolean; results: T[] }
 
-export type SearchQueries = {
+type SearchQueries = {
```
Comment (lukaw3d) [inline src/app/pages/SearchResultsPage/hooks.ts:21]: Is this type still needed?

**Review by lukaw3d — APPROVED** [review]

File: src/app/pages/SearchResultsPage/hooks.ts (hunk 4 above)
Comment (csillag) [reply src/app/pages/SearchResultsPage/hooks.ts:21]: Not really. Removing...

Comment (csillag) [reply src/app/pages/SearchResultsPage/hooks.ts]: Removed

**Pull request merged**